src/konserve/filestore.clj

```diff
@@ -67,8 +67,9 @@
   "Lists all files on the first level of a directory."
   (let [root (Paths/get directory (into-array String []))
         ds (Files/newDirectoryStream root)
+        ephemeral? (fn [^Path path] (starts-with? (.getFileName path) ".nfs"))
```

Comment:
This could be extended to ignore other kinds of temporary files that `konserve` should not care about.

Reply:
Yes, I think it would be good to at least lift the `ephemeral?` predicate to a function argument with your predicate as default. Maybe even better would be a dynamic binding for it, sidestepping the need to lift the argument all up to the surface API.

Reply:
I updated to define and use a dynamic predicate function. Please let me know if you had something else in mind.

I also switched to matching ephemeral files based on regex instead of prefix, for more flexibility; and added a (currently single-element) vector to make an obvious place for adding patterns if needed. I wonder if "ephemeral" might be too limiting a name, but it could be changed if a broader use case appears.

Reply:
Nice work! Sorry for changing my mind again, but I think best would be to actually add `ephemeral?` as a field to the `BackingStore` with your implementation as default value here: 

https://github.com/replikativ/konserve/blob/development/src/konserve/filestore.clj#L690

And then just pass `ephemeral?` to `list-files` as an additional argument in the `-keys` protocol implementation of the `BackingStore` here:

https://github.com/replikativ/konserve/blob/development/src/konserve/filestore.clj#L137

Does this make sense?

Reply:
I think so. Please see the latest diff. I made `list-files` multi-arity, taking `ephemeral?` as an argument in the arity-2 case (so that e.g. `count-konserve-keys` still works as before, making an arity-1 call to `list-files`, as it does its own filtering on `.ksv` filename extensions), and wired up the previous implementation of `ephemeral?` a `BackingFilestore` field. Tests pass.

Reply:
Yes, this is exactly what I meant, thank you! The only thing missing that would be nice is to allow passing in of `ephemeral?` as a keyword argument with your function as default value to `connect-fs-store`.

Reply:
`ephemeral?` is now an optional keyword argument to `connect-fs-store`, with a default implementation. I was able to pass in alternate lambdas to e.g. filter all or no files.

Reply:
Perfect! Thank you!